src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -158,14 +195,39 @@ private async Task CreateBlocksAsync()
                     builder.AppendLine("<<==============================================================>>");
                     this.logger.LogInformation(builder.ToString());
 
-                    await Task.Delay(TimeSpan.FromSeconds((double)this.network.TargetSpacingSeconds / 2.0), this.cancellation.Token).ConfigureAwait(false);
+                    await Task.Delay(TimeSpan.FromSeconds((double) this.network.TargetSpacingSeconds / 2.0), this.cancellation.Token).ConfigureAwait(false);
                 }
                 catch (OperationCanceledException)
                 {
                 }
+                catch (Exception exception)
+                {
+                    this.logger.LogCritical("Exception occurred during mining: {0}", exception.ToString());
+                    break;
+                }
             }
         }
 
+        /// <summary>Gets scriptPubKey from the wallet.</summary>
+        private Script GetPremineScriptPubKey()
```

Comment:
Who owns this wallet?
the best way would be to have the premine sent to a multisig address owned by the members.